### DISCUSSION TITLE
Ingredients and Most Memorable Characteristics

### DISCUSSION THREAD

**Discussion by markusnam** [UBC-MDS/chocolate_rating#25] [General]
These two columns are kind of `comma-delimited`. By viewing the data itself, I think it will be easier to handle `Ingredients`. So let's focus on our 

## Sample data from the raw file without preprocessing
rich cocoa, fatty, bready
cocoa, blackberry, full body
cocoa, vegetal, savory

## Question
How are we going to feed this into our model if we decide to use this feature?

## Option 1: OneHotEncoding (OHE)
- but we cannot apply OHE to this feature directly, otherwise `rich cocoa, fatty, bready` will be treated as one category (correct me if I am wrong)
- then we will likely need to split the feature into multiple features first but the problem is that there are many `characteristics` over the whole dataset... 1038 distinct characteristics in train_df according to EDA result.

## Option 2: CountVectorizer
- I recall that CountVectorizer ignores space and punctuation. It only extracts the words from a string. (correct me I am wrong)
- In this case, can we simply apply CountVectorizer to this `Most Memorable Charteristics` column? The net effect would be similar to using OHE but much simpler in terms of our thinking process and workload (although the code for splitting the column has been developed). And we can make use of `max_features` to ignore those infrequent characteristics.

Personally, if my understanding described above is valid, I would opt for Option 2.

There may be other useful parameters in OHE or CountVectorizer that we may want to explore later.

**Comment by eyrexh** [comment]
I think we can drop the "ingredients" because it is replicated the "most memorable characteristics". I vote for CountVectorizer for "most memorable characteristics" since OHE will generate too many different categories and columns for the analysis.

**Reply by markusnam** [reply]
indeed, ingredients is not an exact replicate of most memorable characteristics but given the fact that almost all ingredient entries come with B (bean) and S (Sugar), it may not help much in terms of rating prediction. Let's drop it for the time being. If we feel that this feature may add some x-factor to our model, we can always come back. :)
